### DISCUSSION TITLE
DeamonSetでのデプロイ方法について

### DISCUSSION THREAD

**Discussion by githubBourbon** [fluent/fluentd#4752] [Q&A (Japanese)] [answered]
### やりたいこと

お世話になります。
fluentdの設定とかに行く前の導入自体の質問になります。
基本過ぎですみませんが宜しくお願い致します

（内容）
３Workerノードの構成で、
OKD(Kubernetes)を構築しています。
すでにPODなどをデプロイ出来る状態になっています。
やりたいことですが、
「fluentdをDaemonSetとして3Workerノードにデプロイをする。」
ということを実施しようとしています。

添付のyamlをapplyしようとしていますが、
fluentdのデータパスは、
ノードのローカルパスをhostPathでマウントさせようとしていますが
これは可能でしょうか？！
Kubernetesのマニュアルだとマルチノードには使えない？！といった
記述がありました。
ただ、DaemonSetとhostPathの事例が多いので
実施させたいのですが、
これを含めてyamlの不足部分がございましたら
よろしくお願い致します。
※　添付のyamlは実行不可でした。
※　fluentdのkubernetesイメージはいくつかGitHUBに上がっていますが
fluentdの送信する先はネットワーク上のストレージ（nfsボリューム）になるのでどのイメージ
を使用すべきか？！になっています。

### 設定した内容

```apache
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: fluent-daemonset
spec:
  selector:
    matchLabels:
      name: fluent-pod
  template:
    metadata:
      labels:
        name: fluent-pod
    spec:
      containers:
        - name: fluent-container
          image: fluentd:latest
          resources:
            requests:
              cpu: 300m
              memory: 300Mi
            limits:
              memory: 500Mi
          volumeMounts:
            - name: config-volume
              mountPath: /fluentd/etc
            - name: var-log
              mountPath: /var/log
            - name: var-log-containers
              mountPath: /var/log/containers
              readOnly: true
      volumes:
        - name: config-volume
          configMap:
            name: fluent-configmap
        - name: var-log
          hostPath:
            　path: /var/log
        - name: var-log-containers
          hostPath:
          　　path: /var/log/containers
```

### 環境について

```markdown
- Fluentd version:
- TD Agent version:
- Fluent Package version:
- Docker image (tag):
- Operating system: OKD 4.15.0
- Kernel version:
- kubectl version:
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.2-3598+6e2789bbd58938-dirty
```

**Comment by daipom** [comment, marked as answer]
@githubBourbon マニフェストファイル(添付のyamlファイル)に全角スペースが含まれているのが原因と思われます。
半角スペースに直して試してみてください。

**Reply by githubBourbon** [reply]
お世話になります。
ありがとうございます。
まぎれてたんですね・・・
それで失敗してたんでしょうか？！
気づきませんでした。
追加で質問したですがこちらでもいいのでしょうか？！
結局、構成としては、３ノードに収集としてFluentBIT，
クラスタ外にForwarder（Fluentd）のLOGサーバになりました。
そこで、
Forwarderでのフィルターについてお聞きしたいです。
いったん別に質問いたします。
ありがとうございます。
